src/components/tooltip/GCMarker.js

```diff
@@ -382,30 +382,37 @@ function _makePhaseTimesArray(
 }
 
 /*
- * Return true if the phase 'phaseName' is a leaf phase among the whole
- * array of phases.
+ * Construct a tree of phase timings, in order to display the most expensive
+ * leaf phases.
  *
- * A leaf phase is a phase with no sub-phases.
+ * A leaf phase is a phase with no sub-phases. The phase tree is constructed
+ * from the phaseTimes array, which contains hierarchical path names of each
+ * phase together with its total elapsed time. We construct synthetic leaf
+ * phases for the self-time of each interior node, containing the elapsed time
+ * that is not part of a sub-phase.
  *
  * If the following are phases:
  * marking.mark_roots
  * marking.mark_heap
- * marking.mark_heap.a
+ * marking.mark_heap.objects
  * sweeping.sweep
  *
- * Then marking.mark_roots, marking.mark_heap.a, sweeping.sweep are the only
- * leaves.  We select these since they will give the person looking at the
- * profile the best clue about which (sub-)phases are taking the longest.
- * For example, it isn't useful to say "sweeping took 200ms" but it is
- * useful to say "sweeping.compacting took 20ms" (if compacting has no
- * sub-phases.
+ * Then marking.mark_roots, marking.mark_heap.objects, sweeping.sweep are the
+ * only original leaves. We will add in "marking.mark_roots (self-time)" and
+ * "marking.mark_heap (self-time)". We select from these leaves since they will
+ * give the person looking at the profile the best clue about which
+ * (sub-)phases are taking the longest.
  *
- * We find leaf phases by constructing a tree of phase times and then
- * reading its leaves.
+ * For example, it isn't useful to say "marking.mark_heap took 200ms" but it is
+ * useful to say "marking.mark_heap.objects took 180ms" (if objects has no
+ * sub-phases). However, if marking.mark_heap took 200ms total yet all of its
+ * sub-phases add up to only 20ms, then we would want to say "marking.mark_heap
+ * (self-time) is 180ms".
```

Comment:
I like your description better than mine, it's clearer with how it works possibly because it's using more familiar ideas.  Your code is also likely to be clearer in this case ;-).  I have only glanced over the code but I can check it if @julienw would like me to.

Reply:
I was mostly updating the comment because (1) the first line seems to refer to a function that no longer exists (at least, I didn't see anything returning true if something is a leaf phase), and (2) to update for the (self-time) change. I didn't intentionally change anything else about the description.